src/components/notifications/NotificationsMenu.js

```diff
@@ -0,0 +1,139 @@
+import React, { useEffect, useState } from "react";
+import { formatDistance, fromUnixTime } from "date-fns";
+import {
+    NOTIFICATIONS_KEY,
+    getNotifications,
+} from "./../../serviceFacades/notifications";
+import { useQuery } from "react-query";
+import { Skeleton } from "@material-ui/lab";
+import NotificationStyles from "./styles";
+import Divider from "@material-ui/core/Divider";
+import ids from "./ids";
+import { build } from "@cyverse-de/ui-lib";
+import Button from "@material-ui/core/Button";
+import {
+    ListItemText,
+    makeStyles,
+    Menu,
+    MenuItem,
+    Typography,
+} from "@material-ui/core";
+import { useTranslation } from "../../i18n";
+
+const useStyles = makeStyles(NotificationStyles);
+
+function getDisplayMessage(notification) {
+    if (notification) {
+        return notification.type === "data"
+            ? notification.subject
+            : notification.message.text;
+    }
+}
+
+function getTimeStamp(time) {
+    if (time) {
+        // slicing because time has extra zeroes in the unix string
+        const d = fromUnixTime(time.slice(0, -3));
+        return formatDistance(d, new Date());
+    }
+}
+
+function NotificationsMenu(props) {
+    const { notificationMssg } = props;
+    const [notifications, setNotifications] = useState([]);
+    const classes = useStyles();
+    const { t } = useTranslation(["common"]);
+
+    const handleClose = () => {
+        props.setAnchorEl();
+    };
+
+    useEffect(() => {
+        if (notificationMssg != null) {
+            setNotifications([notificationMssg, ...notifications].reverse());
+        }
+    }, [notifications, notificationMssg]);
```

Comment:
This will cause infinite re-render (I double checked this by running this PR locally). We should only update the state with the notification message if it is not already in the array.

```suggestion
          const found = notifications?.find(
            (msg) => msg.message.id === notificationMssg?.message.id
        );
        if ((found === null  || found === undefined) && notificationMssg) {
            setNotifications([notificationMssg, ...notifications].slice(0,10));
        }
    }, [notifications, notificationMssg]);
```

Reply:
Also note, my suggestion includes @ash3rz suggestion to limit the state to hold only last 10 notifications at any point-in-time.